### ISSUE TITLE
Nintendont No longer boots game off of usb loader gx

### ISSUE THREAD

**Issue by briantho** [FIX94/Nintendont#347]
Hi, I updated nintendont, and now it no longer works with usb loader gx. Works totally fine when ran from homebrew channel, but no longer works when loading a game from usb loader gx. I've tested this with multiple games, ran from a 750gb usb hdd, and on a real wii. (Not vwii) Every time I load a game it gives a black screren, followed by some hdd activity, then hdd activity stops, then get some error, (It's too fast to read, says something like get game and region failed, or something like that) then usb loader gx reloads. Is there some way to use an old version of nintendont to confirm. I can't seem to figure that out. Also, note a game works totally fine, when loaded directly from nintendont, through the homebrew channel. Using usb loader gx always gives a blank screen, then some error. Again this used to work fine, till I updated nintendont. Please help me figure out this problem.
Thank you, 
Brian

**Comment by briantho** [comment]
ok, error message is 
CheckForMultiGameAndRegion() = failed

**Comment by GerbilSoft** [comment]
Looking into this.

**Comment by GerbilSoft** [comment]
Can't reproduce this here. Using USB Loader GX r1262 and Nintendont v4.430.

What's your exact system setup? I have a USB HDD with a single FAT32 partition, and the game I loaded is usb:/games/GALE01/game.iso (Super Smash Bros. Melee, Rev.00)

**Comment by briantho** [comment]
I have about the same setup. 750gb portable USB HDD, fat32, and the games are under usb:/games/Super Smash Bros Melee [GALE01]/game.iso also, just updated usb loader gx to r1262, was running r1260 but still the same result. Also, got my hdd plugged into the bottem port of the wii, and have cios installed as well as system menu 4.3. Also, I have nintendon't running off the usb hdd as well as usb loader gx (don't know if that matters). Also, have the same nintendont version 4.430. again it works via the homebrew channel fine, just not via usb loader gx anymore...
Let me know if there's anything else I can try. 
Thanks

**Comment by LukeR2016** [comment]
Same setup and same for me, works fine from homebre channel when booting through CFG USB loader takes ages then boots to black screen

**Comment by LukeR2016** [comment]
I should add when it taked ages to load it seems to stick on the ios58 line if that helps

**Comment by LukeR2016** [comment]
ok also tried through USB Loader GX black screen straight away.

**Comment by briantho** [comment]
@LukeR2016 Hi, Glad I'm not the only one having issues, 
@GerbilSoft  Would a log help? If so, how do I create one? also, I just updated nintendont to 4.432, and am getting a different error message almost immediately. It says "No fat device found" IDK what's going on, but it still works fine running off the homebrew channel.

**Comment by briantho** [comment]
ok, so I've discovered something. Switching to a usb flash drive works, however still can't get my portable usb hard drive to work from usb loader gx with nintendont. I'm still getting the "no fat device found", when using usb loader gx. However still works totally fine using nintendont via the homebrew channel. Here's what I tried:
I deleted all settings for nintendont and usb loader gx, to put them in a clean state. just left the boot.dol, meta and icon.png files Result: same problem

Then I tried formatting a flash drive, fat32, 32kb cluster size, and put a few games to test with. Then I copied the usb loader gx and nintnedont apps from my usb hard drive to the usb flash drive. Result: nintendon't worked perfectly from usb loader gx. 

For whatever reason it does not work on my original hdd. Not sure whats causing it, but it still comes up with the "could not find fat device" error. I don't really have another drive to test it out on, and my original hard drive is formatted as fat32, with 32kb cluster size. Is there something that can be done? I don't really want to go buy another drive to test. I'd really like to get this working again. Please help, and thanks in advance. 
Brian

**Comment by carnage702** [comment]
your hdd probably has an hidden partition or something, try reformating it with a formattr that isnt windows.

**Comment by briantho** [comment]
I used easeus partition manager to format originally to fat32, and there are no hidden partitions. Under both easeus and windows disk management, both show no hidden partitions, and it's formatted as fat32. 
However if you think it will help, I'll go ahead and format it. Just need some time to copy stuff off of the drive. I'll report back either way.

**Comment by briantho** [comment]
Got any other ideas? Just formatted my drive as fat32 with easeus partition manager, copied everything back, started usb loader gx, loaded a game, annnnnndddddd.... still the same error. One thing I noticed, when I set usb loader gx's nintendont boot.dol file path to the sd card, I got error  
CheckForMultiGameAndRegion() = failed
However when I changed the boot path to usb, I got error "no fat device found" 
Nintendont still works perfect when loaded via homebrew channel, so basically I'm still back to where I was when I started this post. I still can't figure out why this is happening. It used to work fine with old versions of nintendont, now usb loader gx will no longer work with my hdd and nintnedont. Also, is there any way to enable logging, I enabled it in usb loader gx settings for nintendont, but I never see a log file on my hard drive or sd card. 
Is there anything else I can try?

**Comment by jerseycactus** [comment]
I have nothing else to add to the solution, but I am getting the same error as you, with my 64 GB Flash Drive.

**Comment by jerseycactus** [comment]
Actually, I have figured it out (at least for me). I had to go into the update section, accesses for x on game list, and download md5.  It worked perfectly from that.

**Comment by briantho** [comment]
Ok, so I believe I'm finally heading in the right direction, at least I Think. I discovered that nintendont is somehow incompatible with my usb hard drive enclosure...
Sounds bizarre right? Here's what I did:
SO, I took apart my aftermarket usb 2.0 hard drive enclosure. I removed the 2.5 laptop drive, and connected it to a external usb to sata adapter, then plugged it into my wii. With the drive connected to the usb to sata adapter, nintendont worked perfectly with usb loader gx! Amazed, I tried a few games, all worked perfectly. I then tried connecting the drive back to my old enclosure, and tried the same thing. It failed. Tried again with the usb to sata adapter, and worked perfect. Somehow my old drive enclosure became incompatible with nintendont, even though it worked when booted off the homebrew channel. Very Weird. Anyway, I was just about to go and buy a another usb enclosure to try, to see if it will work. If it does I'll go ahead and close this issue.

**Comment by LukeR2016** [comment]
I think your correct could be a speed issue. I used a Tosh USB all ok. My Sandisk fails, read somewhere its to do with the speed of the drive!

**Comment by GerbilSoft** [comment]
It may be some sort of (re-)initialization issue with the drive case when chainloading from USB Loader GX to Nintendont. I don't think disk speed would be causing this problem.

This would also explain why I can't reproduce it with my HDD.

**Comment by briantho** [comment]
Allright, so Getting a [new enclosure](https://www.amazon.com/gp/product/B00JQTO8TU/) has solved my issue, nintendon't now works perfectly via usb loader gx. I think @GerbilSoft is right and my old enclosure was having some sort of weird re-initialization issue. I'm going to go ahead and close this issue, and if helps anyone out in the future, the model of my new enclosure is a inateck FE2004. Thanks again for the support guys, 
Brian

**Comment by GerbilSoft** [comment]
I have access to an FE2005 enclosure. I'm not sure if it has the same chip, but I can test it out on USB Loader GX later. (I know it definitely works with Nintendont by itself; I haven't tested ULGX chainloading.)

**Comment by briantho** [comment]
Hi, the FE2004 is the working enclosure, the model of the non-working enclosure is Vantec NexStar TX USB 2.0 Enclosure. Also, I have a hunch the issue may have started when nintendon't stopped displaying the nintendon't loading info (the screen that shows nintendon't loading, right after selecting the game via usb loader gx, but before the game actually boots). Basically what I've noticed is as of recent versions of nintendont, is that when you select a game via usb loader gx, it gives a blank screen followed by the game booting (No nintendon't loading screen). It seems to me that may be the difference, as to why older versions of nintendont worked fine with my NexStar TX, and newer versions of nintendont don't work. It might also explain why it was working with newer versions of nintendont via the homebrew channel directly. Booting via the homebrew channel directly gives the nintendont diagnostic loading screen. Anyway, if you can find access to a nexstar tx, or something similar, feel free to debug. Thanks 
Brian

**Comment by wellspokenman** [comment]
Just FYI: I have a 500GB fat32 (exfat) drive connected to usb2:/ that works with USB Loader GX for Wii games, but did not when USB-GX tried to call Nintendont for GC ISOs. This error went away for me after removing the flash drive from usb1:/, but I have no idea why.

**Comment by DaveGerrist** [comment]
Hi, same error here: CheckForMultiGameAndRegion() = failed
 
I'm using a sd card (32GB) with loader GX and Nintendont on it. After installing the loader Gx, Nintendont gave me this error. I was not able to load Gamecube games from loader Gx, so i've changed the Loader settings to Nintendont:
Loader Gx -> Settings -> Loader settings -> Gamecube Mode.

Then you might want to check the Gamecube Usb and SD path settings:
Loader Gx -> Settings -> Custom path settings

If you still have this issue please check your SD Card lock switch on the side of the card. When it's locked Nintendont will give this error and you will also not be able to load Gamecube games with Loader GX.

**Comment by donut3226** [comment]
I have the same issue/solution as wellspokenman - I know this thread is a little old, but only thing I could find online related to my issue.

Funny thing is, the HDD was working with Wii and GC games until recently.  Not sure what triggered the change.  

Removing my other HDD (hosting wii u games) resolves the issue for me too.

**Comment by GRUMPYx87** [comment]
I know this thread is almost a year old but im having sort of the same issue, my enclosure isn’t compatible with the wii (NOT Vwii) 
Both the usbloadergx and Nintendont dont work.. as for the usbloadergx some games give me a black screen also there’s massive lag on cutscenes.. already tried two different hdd and same issue.. can anyone recommend a compatible enclosure for a 2.5 sata hdd??

**Comment by pawhtiobo** [comment]
This is a kind of trial-error situation, there are a lot of lists in the internet with some recommended enclosures, but they are a bit outdated, you can check them, but its not guaranteed that it will work in your Wii, every Wii is different, my first recommendations would be:

1- Check if the HDD correctly partitioned and is formatted in FAT32 with 32k clusters format (very important).
2- Use a tool like CrystalDiskInfo and disable all AAM and APM (Menu Function - Advanced Features - AAM/APM Control) of the HDD.

**Comment by GRUMPYx87** [comment]
UPDATE: Just received my new enclosure today, its a inateck model (FE2004) 
Wii/Gamecube games run perfect.. no lag ect..

**Comment by sneakypete76** [comment]
I can confirm this. Certain enclosures produce the problem and extra devices connected to the second USB port (ie, flash drives and harddrives). None of my games would run on my HDD on port 1, until I unplugged the additional flashdrive from port 2, then the games launched correctly. Seems to be a voltage issue which is strange.

**Comment by millenius** [comment]
What IOS base are you all using? For my inateck fe2004 USB Loader vs won't see my Wii games and chainloading doesnt work either I can only load GC games through opening nintendont from HBC

**Comment by razerocksxd** [comment]
This helped me: If you have memory cards plugged in but Nintendont tries to use emulated memory cards, it will result in a black screen. Either remove memory cards or change settings.

**Comment by moniker-supreme** [comment]
I solved this error for my specific situation: I got a "CheckForMultiGameAndRegion() failed: 1" trying to boot Gamecube titles via USBLoaderGX after replacing my flash drive with a new one of a higher capacity/read speed.

My scenario was: I had just bought the new, bigger USB flash drive, formatted to FAT32, copied over my files/folders from the previous & working 256gb to the new 512gb (same model, SanDisk USB 3.1), and even though the files were identical, GC roms wouldn't boot via USBLoaderGX on the new flash drive.

They boot from the Nintendont app just fine, however. This indicated to me that the USB wasn't necessarily incompatible or corrupted, but that USBLoaderGX wasn't reading from it correctly. After searching and skimming threads like this, I almost gave up and returned my new flash drive for a new one thinking it was incompatible.

But, after some messing around, I found the fix was simply changing the setting "Remove Disk Read Speed Limit" within USBLoaderGX from "OFF" to "ON". It seems that my USB read speed is faster than expected for loading GC ISO's. Note: This option turned "ON" may cause errors for certain games that are sensitive to read speeds. If you have errors, try a USB 2.0 or 3.0 device instead. Hope this helps someone.
